base/intfuncs.jl

```diff
@@ -325,7 +325,8 @@ end
 
 function digits!{T<:Integer}(a::AbstractArray{T,1}, n::Integer, base::Integer=10)
     2 <= base || throw(ArgumentError("base must be ≥ 2, got $base"))
-    base - 1 <= typemax(T) || throw(ArgumentError("type $T too small for base $base"))
+    !applicable(typemax, T) ||
```

Comment:
Oh, I didn't expect that to be a performance problem, but still it seems to be reasonable (of the order of 30 percent penalty for small numbers). I just tried with `method_exists` instead, but this is far worse. I don't like the `T == BigInt` solution, as it doesn't handle number types defined outside of Base.